### ISSUE TITLE
Question about the updata of the synthetic data

### ISSUE THREAD

**Issue by Ke-Hao** [ssnl/dataset-distillation#60]
Hi, I found these codes confusing : [codes](https://github.com/ssnl/dataset-distillation/blob/master/train_distilled_image.py#L149-L153)
why the outputs is 'gw' , and the synthetic data is updated in the accumulate_grad() , why there is a optimizer.step() in the function train()?

**Comment by ssnl** [comment]
+ Consider the computation graph for each GD step. `gw` is the output of that.

+ I don't see synthetic data being updated in `accumate_grad()`, which simply gathers gradients (of synthetic data) from all GD steps.

+ `optimizer.step()` updates the synthetic data according to the distillation loss. So it is naturally called in `train()`, which trains the synthetic data.

**Comment by Ke-Hao** [comment]
Got it! Thanks for your reply!!
